tests/components/totalconnect/test_diagnostics.py

```diff
@@ -0,0 +1,20 @@
+"""Test TotalConnect diagnostics."""
+
+from homeassistant.components.diagnostics import REDACTED
+
+from .common import DEVICE_INFO_BASIC_1, LOCATION_ID, init_integration
+
+from tests.components.diagnostics import get_diagnostics_for_config_entry
+
+
+async def test_entry_diagnostics(hass, hass_client):
```

Comment:
Could this test be expanded to include user, partitions etc?

Or even return a whole example diagnostic tree to verify it is represented as you were expecting?

E.g. https://github.com/home-assistant/core/blob/dev/tests/components/airvisual/test_diagnostics.py

Reply:
I added a test for each item type (device, partition, etc).